src/sql/platform/connection/common/constants.ts

```diff
@@ -65,3 +65,6 @@ export const UNSAVED_GROUP_ID = 'unsaved';
 /* Server Type Constants */
 export const sqlDataWarehouse = 'Azure SQL Data Warehouse';
 export const gen3Version = 12;
+
+/* Password Reset Error code */
+export const passwordErrorCode = 18488;
```

Comment:
This is specific to MSSQL - but the feature itself seems generic to all providers. Is the intent for this to be something that all providers can get automatically? If so you're going to need to refactor this to have the provider be able to tell you when an error is specifically because a password change is needed.

Reply:
For now the feature is only compatible with SqlServer as the change password function is part of the SqlClient library (seems to be meant for Sql Servers)

Reply:
Well, that's just an implementation detail. Providers would be the one actually doing the calls (which in the case of MSSQL is STS) - but other providers could theoretically want to hook into this and have their own implementation called when needed using whatever method of changing the password is available to them.

Reply:
Yeah true, but I'm not aware how any other provider uses the change password feature. For now, I'm focused on Sql Server (though I will need to modify the provider class to have the password change needed indicator for another time).